test/box-luatest/gh_8817_bug_set_language_shared_test.lua

```diff
@@ -0,0 +1,80 @@
+local t = require('luatest')
+local fio = require('fio')
+local popen = require('popen')
+local server = require('luatest.server')
+
+local g = t.group()
+
+-- Create server test instance.
+g.before_all(function()
+    g.server = server:new({alias = 'test-gh-8817-server'})
+    g.server:start()
+    -- Get port for clients.
+    g.server_port = g.server:exec(function()
+        return require('console').listen(0):name()['port']
+    end)
+end)
+
+-- Stop server test instance.
+g.after_all(function()
+    g.server:stop()
+end)
+
+-- Checks that the language setting is not shared between clients.
+g.test_set_language_not_shared_between_clients = function()
+    -- Find tarantoolctl.
+    -- Probably test-run knows the path to tarantoolctl.
+    local TARANTOOLCTL_PATH = os.getenv('TARANTOOLCTL')
+    if TARANTOOLCTL_PATH == nil then
+        -- If test-run dose not know the path to tarantoolctl.
+        -- Assume that current directory is tarantool's build directory.
+        local BUILDDIR = os.getenv('BUILDDIR')
+        TARANTOOLCTL_PATH = ('%s/extra/dist/tarantoolctl'):format(BUILDDIR)
+        if not fio.path.exists(TARANTOOLCTL_PATH) then
+            error("Can't find tarantoolctl")
+        end
+    end
+
+    local cmd_tarantoolctl_connect = {
+        TARANTOOLCTL_PATH,
+        'connect',
+        tostring(g.server_port),
+    }
+
+    -- First connection will be established in a separate process.
```

Comment:
Is it worth to check that `box.session.id` are different for two clients?

Reply:
Done, added